File: bindings/ruby/tests/cim_enumerate_all.rb
# cim_enumerate_classnames.rb
#  test EnumerateClassnames CIM extension

$:.unshift "../../../build/bindings/ruby"
$:.unshift "../.libs"

require 'test/unit'
require 'rexml/document'
require 'openwsman'
require '_client'

class WsmanTest < Test::Unit::TestCase
  def test_enumerate_classnames
#    Openwsman::debug = 1
    client = Client.open
    assert client
    options = Openwsman::ClientOptions.new
    assert options
    options.set_dump_request
    uri = "http://schemas.dmtf.org/wbem/wscim/1/*"
#    uri = "http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/CIM_HostedService"
#    uri = "http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/CIM_CommMechanismForManager"
#    uri = "http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/CIM_ManagedElement"
    result = client.enumerate( options, nil, uri )
    assert result
#    puts result

    count = 0
    context = result.context
    while (context)
	count += 1
	result = client.pull( options, nil, uri, context )
	puts "Response ##{count}"
#	puts "### #{result}"
#	items = result.body.PullResponse.Items.child
	items = result.body["PullResponse"]
	assert items
	items = items["Items"]
	assert items
	items = items.first
	assert items
	items.each { |child|
	    puts "\t#{child.name} = #{child.text}"
	}
	context = result.context
    end
    client.release( options, uri, context ) if context
    puts "Got #{count} responses"
  end
end

File: bindings/java/tests/cim_enumerate_all.java
import org.openwsman.Client;
import org.openwsman.ClientOptions;
import org.openwsman.XmlDoc;
import org.openwsman.jwsmanConstants;

public class cim_enumerate_all {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		Client client = new Client(
				"http://wsman:secret@linux-3qev.localnet:5985/wsman");
		ClientOptions options = new ClientOptions();
		// op.set_dump_request();
		client.transport().set_auth_method("basic");

		System.err.println("Enumerate");
		XmlDoc result = client.enumerate(options, null, jwsmanConstants.CIM_ALL_AVAILABLE_CLASSES);
		if ((result == null) || (result.is_fault() != 0))
			System.err.println("Enumeration failed");
		else {
			String context = result.context();
			System.err.println("Context: " + context);
			while (context != null) {
				System.err.println("Pull");
				result = client.pull(options, null, jwsmanConstants.CIM_ALL_AVAILABLE_CLASSES, context);
				if (result != null)	 {
					System.out.println(result.encode("UTF-8"));
					context = result.context();
				} else
					context = null;
			}
		}
	}

}
